### ISSUE TITLE
[a11y]: RadioButtonGroup has invalid aria-readonly attribute

### ISSUE THREAD

**Issue by finken2** [carbon-design-system/carbon#13665]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.25.0

### React version

v16.12.0

### Automated testing tool and ruleset

Accessibility Checker in chrome

### Assistive technology

n/a

### Description

Setting a RadioButtonGroup to readonly sets `aria-readonly` on the fieldset which is invalid. This can be seen in the carbon-react storybook. 

From accessibility checker:
The ARIA attributes "aria-readonly" are not valid for the element <fieldset> with implicit ARIA role "group"

### WCAG 2.1 Violation

4.1.2

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-radiobutton--playground&args=readOnly:true

### Steps to reproduce

radioButtonGroup in storybook, run accessibility checker. You may need to flip readonly to true in the playground. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
